src/Platforms/AbstractPlatform.php

```diff
@@ -872,7 +872,7 @@ private function buildCreateTableSQL(Table $table, bool $createForeignKeys): arr
     }
 
     /**
-     * @param list<Table> $tables
+     * @param array<Table> $tables
```

Comment:
This is less specific not more, but running this locally, I can see why it is needed: this has ripple effects throughout the codebase.

Reply:
There are a lot of questionable changes in this PR when it comes to array types, and I won't be able to explain may of them. I can only tell that PHPStan (given the current settings) and Psalm handle them quite differently. For instance, for Psalm `list<T>` is a subtype of `array<int<0,max>,T>` but for PHPStan it's not. So there are a lot of compromises here to satisfy them both.